provisioner/terraform/provision_test.go

```diff
@@ -82,7 +82,21 @@ func makeTar(t *testing.T, files map[string]string) []byte {
 	t.Helper()
 	var buffer bytes.Buffer
 	writer := tar.NewWriter(&buffer)
+
+	addedDirs := make(map[string]bool)
 	for name, content := range files {
+		// Add parent directories if they don't exist
+		dir := filepath.Dir(name)
+		if dir != "." && !addedDirs[dir] {
+			err := writer.WriteHeader(&tar.Header{
+				Name:     dir + "/", // Directory names must end with /
+				Mode:     0o755,
+				Typeflag: tar.TypeDir,
+			})
+			require.NoError(t, err)
+			addedDirs[dir] = true
+		}
```

Comment:
I found many instances of this function sprinkled around our codebase so I added a helper function for this in https://github.com/coder/coder/pull/15540

I'll wait for this to be merged and then update it to use `testutil.CreateTar` :+1: